lib/jmap/core/session/session.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:equatable/equatable.dart';
+import 'package:jmap_dart_client/http/converter/account_converter.dart';
+import 'package:jmap_dart_client/http/converter/account_id_converter.dart';
+import 'package:jmap_dart_client/http/converter/capabilities_converter.dart';
+import 'package:jmap_dart_client/http/converter/state_converter.dart';
+import 'package:jmap_dart_client/http/converter/user_name_converter.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/account/account.dart';
+import 'package:jmap_dart_client/jmap/core/capability/capability_identifier.dart';
+import 'package:jmap_dart_client/jmap/core/capability/capability_properties.dart';
+import 'package:jmap_dart_client/jmap/core/state.dart';
+import 'package:jmap_dart_client/jmap/core/user_name.dart';
+
+class Session with EquatableMixin {
+
+  final Map<CapabilityIdentifier, CapabilityProperties> capabilities;
+  final Map<AccountId, Account> accounts;
+  final Map<CapabilityIdentifier, AccountId> primaryAccounts;
+  final UserName username;
+  final Uri apiUrl;
+  final Uri downloadUrl;
+  final Uri uploadUrl;
+  final Uri eventSourceUrl;
+  final State state;
+
+  Session(
+    this.capabilities,
+    this.accounts,
+    this.primaryAccounts,
+    this.username,
+    this.apiUrl,
+    this.downloadUrl,
+    this.uploadUrl,
+    this.eventSourceUrl,
+    this.state,
+  );
+
+  factory Session.fromJson(Map<String, dynamic> json, CapabilitiesConverter converter) {
+    return Session(
+      (json['capabilities'] as Map<String, dynamic>)
```

Comment:
no, if it is null, exception need to be thrown, and application should be handle error because session is not valid.
please check: https://jmap.io/spec-core.html#the-jmap-session-resource